kvbc/src/v4blockchain/detail/latest_keys.cpp

```diff
@@ -236,6 +236,80 @@ void LatestKeys::trimHistoryUntil(BlockId block_id) {
   }
 }
 
+std::optional<categorization::Value> LatestKeys::getValue(BlockId latest_block_id,
+                                                          const std::string& category_id,
+                                                          const std::string& latest_version,
+                                                          const std::string& key) const {
+  std::string get_key;
+  std::string out_ts;
+  const auto& prefix = category_mapping_.categoryPrefix(category_id);
+  get_key.append(prefix);
+  get_key.append(key);
+  auto opt_val = native_client_->get(v4blockchain::detail::LATEST_KEYS_CF, get_key, latest_version, &out_ts);
+  if (!opt_val) {
+    return std::nullopt;
+  }
+  const size_t total_val_size = opt_val->size();
+  const auto* serialized_start = reinterpret_cast<uint8_t*>(opt_val->data());
+  const auto* serialized_end = reinterpret_cast<uint8_t*>(opt_val->data() + (total_val_size - FLAGS_SIZE));
+
+  switch (category_mapping_.categoryType(category_id)) {
+    case concord::kvbc::categorization::CATEGORY_TYPE::block_merkle:
+      return categorization::MerkleValue{{latest_block_id, opt_val->substr(0, total_val_size - FLAGS_SIZE)}};
```

Comment:
the return block id should be ```out_ts``` i.e. the real version of the kv

Reply:
thanks @cloudnoize , changing this.